### DISCUSSION TITLE
Add option to make operators new and delete weak for stack-only implementations

### DISCUSSION THREAD

**Discussion by asmfreak** [modm-io/modm#746] [Ideas]
In a stack-only projects, it is often hard to catch which part of a project is actually allocating data.
Most of the time usage of C-style allocation (`malloc`, `free` and friends) can be relatively easy debugged.
The problem is C++ operators, which in modm implementation just call those C-style functions. They can be called from a very surprising contexts, like in virtual destructors see [this post on SO](https://stackoverflow.com/questions/31686508/why-is-delete-operator-required-for-virtual-destructors). 

Even if the application as a whole does not use allocations, such smaller issues can force one to include heap. So I propose to add a weak attribute depending on a lbuild switch

**Comment by salkinium** [comment]
Sure, go for it! I don't even think it needs to be optional. Perhaps there's also a compile flag that disables this behavior completely?
